### PR TITLE
add push image after build/test

### PR DESCRIPTION
**Pull request by ar3ndt**

### CODE REVIEW

**Review by mkyc — CHANGES_REQUESTED** [review]
This logic should be divided into two separate jobs exactly like in here: 

https://github.com/epiphany-platform/m-azure-kubernetes-service/blob/x/mkyc/feature/29-subnet-tests/pipeline.yml

**Comment by ar3ndt** [comment]
> This logic should be divided into two separate jobs exactly like in here:
> 
> https://github.com/epiphany-platform/m-azure-kubernetes-service/blob/x/mkyc/feature/29-subnet-tests/pipeline.yml

ready

File: pipeline.yml [hunk 1]
```diff
@@ -8,54 +8,88 @@ pr:
 pool:
   name: $(poolName)
 
-
-variables:
-  ver: dev
```
Comment (mkyc) [inline pipeline.yml:13]: this should be used ...

File: pipeline.yml [hunk 2]
```diff
@@ -8,54 +8,88 @@ pr:
...
+  - job: Build_and_Test
+    steps:
+      - task: Bash@3
+        displayName: 'Set Build Variables'
+        inputs:
+          targetType: 'inline'
+          script: |
+            echo "##vso[task.setvariable variable=MAJOR_VERSION]$(make print-VERSION)"
+            echo "##vso[task.setvariable variable=IMAGE_NAME]$(make print-IMAGE_NAME)"
+            echo "##vso[task.setvariable variable=IMAGE_REPOSITORY]$(make print-IMAGE_REPOSITORY)"
```
Comment (mkyc) [inline pipeline.yml:21]: and I don't think you would need to scrap it here from makefile.

File: pipeline.yml [hunk 3]
```diff
@@ -8,54 +8,88 @@ pr:
...
+          failOnStderr: true
+        env:
+          K8S_VOL_PATH: '/tmp/tests-share'
+          K8S_HOST_PATH: '/tests-share'
+          CGO_ENABLED: 0
+          AZURE_CLIENT_ID: $(clientId)
+          AZURE_CLIENT_SECRET: $(clientSecret)
+          AZURE_SUBSCRIPTION_ID: $(subscriptionId)
+          AZURE_TENANT_ID: $(tenantId)
+          VERSION: $(MAJOR_VERSION)
```
Comment (mkyc) [inline pipeline.yml:62]: .... and you would use ver variable here.

File: pipeline.yml [hunk 4]
```diff
@@ -8,54 +8,88 @@ pr:
...
+      - task: Bash@3
+        displayName: 'go test'
+        inputs:
+          targetType: 'inline'
+          script: |
+            go env
+            make build
+            make pipeline-test | tee tests.output
+            go-junit-report < tests.output > report.xml
+            docker tag $(IMAGE_REPOSITORY):$(MAJOR_VERSION) $(registry_name)/$(IMAGE_REPOSITORY):$(MAJOR_VERSION).$(Build.BuildId)
```
Comment (mkyc) [inline pipeline.yml:52]: I don't understand why would we tag image here if it's done automatically in line 80

File: pipeline.yml [hunk 5]
```diff
@@ -8,54 +8,88 @@ pr:
...
+          tags: $(Build.BuildId)
+          Dockerfile: $(System.DefaultWorkingDirectory)/Dockerfile
+          buildContext: $(System.DefaultWorkingDirectory)
+      - task: CmdLine@2
+        displayName: 'Cleanup Local Image'
+        inputs:
+          script: |
+            echo "Removing image"
+            echo "---------------------------------------------------------------"
+            docker rmi $(registry_name)/$(IMAGE_REPOSITORY):$(Build.BuildId)
```
Comment (mkyc) [inline pipeline.yml:95]: add new line

**Review by mkyc — CHANGES_REQUESTED** [review]
I believe this file should be mirror of already mentioned: https://github.com/epiphany-platform/m-azure-kubernetes-service/blob/x/mkyc/feature/29-subnet-tests/pipeline.yml If there is good reason to improve this file then ok, but I believe we would like to keep them as similar as possible. 

Additional comment is that I believe that it's CI/CD variables (ver or clientId, clientSecret, subscriptionId, and so on ...) should go first, so I'd rather avoid snapping values from Makefile. I believe even [this](https://github.com/epiphany-platform/m-azure-kubernetes-service/blob/x/mkyc/feature/29-subnet-tests/pipeline.yml#L77) should be removed eventually.

File: pipeline.yml (hunk 1 above)
Comment (ar3ndt) [reply pipeline.yml:13]: why? we use variable set in makefile so no point to have it configured in several places.

File: pipeline.yml (hunk 2 above)
Comment (ar3ndt) [reply pipeline.yml:21]: why not, it is better approach then hardcoding variable.

File: pipeline.yml (hunk 3 above)
Comment (ar3ndt) [reply pipeline.yml:62]: as above

File: pipeline.yml (hunk 4 above)
Comment (ar3ndt) [reply pipeline.yml:52]: tagging is leftover by previous version, ill remove it

**Comment by mkyc** [comment]
As I wrote in PR comment: 

> (...) I believe that it's CI/CD variables (ver or clientId, clientSecret, subscriptionId, and so on ...) should go first, so I'd rather avoid scrapping values from Makefile (...)

**Review by mkyc — DISMISSED** [review]
LGTM

**Review by atsikham — APPROVED** [review]

**Review by mkyc — APPROVED** [review]

**Review by atsikham — APPROVED** [review]

**Pull request merged**